pkg/cmd/server/api/validation/oauth.go

```diff
@@ -160,6 +164,18 @@ func ValidateLDAPIdentityProvider(provider *api.LDAPPasswordIdentityProvider) Va
 	return validationResults
 }
 
+// RemoteConnection fields validated separately -- this is for keystone-specific validation
+func ValidateKeystoneIdentityProvider(provider *api.KeystonePasswordIdentityProvider, identityProvider api.IdentityProvider) ValidationResults {
+	validationResults := ValidationResults{}
+
+	if !strings.HasPrefix(provider.RemoteConnectionInfo.URL, "https") {
+		validationResults.AddWarnings(fielderrors.NewFieldInvalid("provider.url", provider.RemoteConnectionInfo.URL, "Auth URL should be secure and start with https"))
+		return validationResults
```

Comment:
I don't see anything obvious listed in the keystone API docs around domain name formatting, but we should make sure it's present and probably validate the URL against v3 (and call AuthenticateV3).